addons/account_check_printing/models/account_payment.py

```diff
@@ -249,7 +249,7 @@ def _check_make_stub_pages(self):
         self.ensure_one()
 
         def prepare_vals(invoice, partials=None, current_amount=0):
-            number = ' - '.join([invoice.name, invoice.ref] if invoice.ref else [invoice.name])
+            number = ' - '.join([invoice.name, invoice.ref] if invoice.ref else [invoice.name or '/'])
```

Comment:
It will still crash with your fix if a Bill Reference is set on the bill.

```suggestion
            invoice_name = invoice.name or '/'
            number = ' - '.join([invoice_name, invoice.ref] if invoice.ref else [invoice_name])
```